lib/ansible/modules/web_infrastructure/ansible_tower/tower_organization.py

```diff
@@ -0,0 +1,141 @@
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: tower_organization
+version_added: "2.3"
+short_description: create, update, or destroy Ansible Tower organizations
+description:
+    - Create, update, or destroy Ansible Tower organizations. See
+      U(https://www.ansible.com/tower) for an overview.
+options:
+    name:
+      description:
+        - Name string to use for the organization.
+      required: True
+      default: null
+    description:
+      description:
+        - String to use for the organization.
+      required: False
+      default: null
+    state:
+      description:
+        - Desired state of the resource.
+      required: False
+      default: "present"
+      choices: ["present", "absent"]
+    config_file:
+      description:
+        - Path to the Tower config file. See notes.
+      required: False
+      default: null
+
+
+requirements:
+  - "python >= 2.6"
+  - "ansible-tower-cli >= 3.0.3"
+
+notes:
+  - If no I(config_file) is provided we will attempt to use the tower-cli library
+    defaults to find your Tower host information.
+  - I(config_file) should contain Tower configuration in the following format:
+      host=hostname
+      username=username
+      password=password
+'''
+
+
+EXAMPLES = '''
+  - tasks
+    - name: Create organization
+      tower_organization:
+          name: "Foo"
+          description: "Foo bar organization"
+          state: present
+          config_file: "~/tower_cli.cfg"
+'''
+
+import os
+
+try:
+    import tower_cli
+    import tower_cli.utils.exceptions as exc
+    from tower_cli.utils import parser
+    from tower_cli.conf import settings
+
+    HAS_TOWER_CLI = True
+except ImportError:
+    HAS_TOWER_CLI = False
+
+
+def tower_auth_config(module):
+    config_file = module.params.get('config_file')
+    if not config_file:
+        return {}
+
+    config_file = os.path.expanduser(config_file)
+    if not os.path.exists(config_file):
+        module.fail_json(msg='file not found: %s' % config_file)
+    if os.path.isdir(config_file):
+        module.fail_json(msg='directory can not be used as config file: %s' % config_file)
+
+    with open(config_file, 'rb') as f:
+        return parser.string_to_dict(f.read())
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            name = dict(),
+            description = dict(),
+            config_file = dict(),
+            state = dict(choices=['present', 'absent'], default='present'),
+        ),
+        supports_check_mode=False
```

Comment:
We would have to encode a lot of logic in the module to properly support check mode. And even doing that, we still couldn't support it 100% for destructive changes (we won't know if they are denied until we try it). A user may be able to see/read an organization and not update it. We wouldn't know that until they attempted to update the organization, so check mode would report it would be changed/updated and then running would result in a permission issue.

We also have the issue of, if I query for an organization that I don't have permission to, we return a 404 Not Found. No where does that give me the information about my ability to create that organization, so for check mode we would return that a change/udpate would happen, yet I may get permission denied at runtime.

At best, check mode for this module would have to speculate about the potential outcome, probably get it wrong pretty often, and will create a worse user experience. For those reasons I don't think supporting check mode for this module is a good idea.